### PR TITLE
fix NullPointerException in listing VMScaleSets and VMs of VMScaleSet

### LINKED ISSUE #25231: [BUG] NullPointerException in listing VMScaleSets and VMs of VMScaleSet
**Describe the bug**
 We are getting NullPointerException in listing VMs. This is happing when we are listing VMScaleSets and VMs of VMScaleSet 

***Exception or Stack Trace***

java.lang.NullPointerException: null
	at com.azure.resourcemanager.compute.implementation.VirtualMachineScaleSetImpl.initializeChildrenFromInner(VirtualMachineScaleSetImpl.java:183) ~[azure-resourcemanager-compute-2.8.0.jar!/:2.8.0]
	at com.azure.resourcemanager.resources.fluentcore.arm.models.implementation.GroupableParentResourceImpl.<init>(GroupableParentResourceImpl.java:32) ~[azure-resourcemanager-resources-2.8.0.jar!/:2.8.0]
	at com.azure.resourcemanager.compute.implementation.VirtualMachineScaleSetImpl.<init>(VirtualMachineScaleSetImpl.java:169) ~[azure-resourcemanager-compute-2.8.0.jar!/:2.8.0]
	at com.azure.resourcemanager.compute.implementation.VirtualMachineScaleSetsImpl.wrapModel(VirtualMachineScaleSetsImpl.java:238) ~[azure-resourcemanager-compute-2.8.0.jar!/:2.8.0]
	at com.azure.resourcemanager.compute.implementation.VirtualMachineScaleSetsImpl.wrapModel(VirtualMachineScaleSetsImpl.java:32) ~[azure-resourcemanager-compute-2.8.0.jar!/:2.8.0]
	at com.azure.resourcemanager.resources.fluentcore.arm.collection.implementation.ReadableWrappersImpl.lambda$wrapPageAsync$1(ReadableWrappersImpl.java:32) ~[azure-resourcemanager-resources-2.8.0.jar!/:2.8.0]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[?:1.8.0_292]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384) ~[?:1.8.0_292]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[?:1.8.0_292]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:1.8.0_292]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[?:1.8.0_292]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_292]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566) ~[?:1.8.0_292]
	at com.azure.resourcemanager.resources.fluentcore.utils.PagedConverter.lambda$mapPagedResponse$6(PagedConverter.java:111) ~[azure-resourcemanager-resources-2.8.0.jar!/:2.8.0]
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:106) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxTake$TakeSubscriber.onNext(FluxTake.java:128) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxExpand$ExpandBreathSubscriber.onNext(FluxExpand.java:118) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onNext(FluxSwitchIfEmpty.java:74) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxMap$MapConditionalSubscriber.onNext(FluxMap.java:220) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onNext(MonoFlatMap.java:249) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onNext(FluxSwitchIfEmpty.java:74) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onNext(MonoFlatMap.java:249) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onNext(MonoFlatMap.java:249) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2398) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onSubscribe(MonoFlatMap.java:238) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:55) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:157) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onNext(FluxSwitchIfEmpty.java:74) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2398) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set(Operators.java:2194) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onSubscribe(Operators.java:2068) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:55) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:64) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:157) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoCacheTime$CoordinatorSubscriber.signalCached(MonoCacheTime.java:337) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoCacheTime$CoordinatorSubscriber.onNext(MonoCacheTime.java:354) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:151) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2398) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onSubscribe(MonoFlatMap.java:110) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:55) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:64) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoCacheTime.subscribeOrReturn(MonoCacheTime.java:143) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:57) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:157) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.onNext(FluxDoOnEach.java:173) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:151) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:151) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:151) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.complete(MonoIgnoreThen.java:284) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onNext(MonoIgnoreThen.java:187) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:151) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoCreate$DefaultMonoSink.success(MonoCreate.java:160) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at com.azure.core.http.okhttp.OkHttpAsyncHttpClient$OkHttpCallback.onResponse(OkHttpAsyncHttpClient.java:181) ~[azure-core-http-okhttp-1.7.3.jar!/:1.7.3]
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519) ~[okhttp-4.9.1.jar!/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
	Suppressed: java.lang.Exception: #block terminated with an error
		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:99) ~[reactor-core-3.4.9.jar!/:3.4.9]
		at reactor.core.publisher.Flux.blockLast(Flux.java:2644) ~[reactor-core-3.4.9.jar!/:3.4.9]
		at com.azure.core.util.paging.ContinuablePagedByIteratorBase.requestPage(ContinuablePagedByIteratorBase.java:94) ~[azure-core-1.20.0.jar!/:1.20.0]
		at com.azure.core.util.paging.ContinuablePagedByItemIterable$ContinuablePagedByItemIterator.<init>(ContinuablePagedByItemIterable.java:50) ~[azure-core-1.20.0.jar!/:1.20.0]
		at com.azure.core.util.paging.ContinuablePagedByItemIterable.iterator(ContinuablePagedByItemIterable.java:37) ~[azure-core-1.20.0.jar!/:1.20.0]
		at com.azure.core.util.paging.ContinuablePagedIterable.iterator(ContinuablePagedIterable.java:106) ~[azure-core-1.20.0.jar!/:1.20.0]
		at io.avidsecure.cloud.azure.api.vmscaleset.AzureVMScaleSetFetcher.getScaleSets(AzureVMScaleSetFetcher.java:65) ~[classes!/:0.0.1-SNAPSHOT]
		at io.avidsecure.cloud.azure.api.vmscaleset.AzureVMScaleSetFetcher.fetch(AzureVMScaleSetFetcher.java:49) [classes!/:0.0.1-SNAPSHOT]
		at io.avidsecure.cloud.Fetcher.checkAndFetchData(Fetcher.java:97) [classes!/:0.0.1-SNAPSHOT]
		at io.avidsecure.cloud.Fetcher.checkAndFetch(Fetcher.java:72) [classes!/:0.0.1-SNAPSHOT]
		at io.avidsecure.cloud.Fetcher$$FastClassBySpringCGLIB$$d7f67faf.invoke(<generated>) [classes!/:0.0.1-SNAPSHOT]
		at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) [spring-core-5.3.9.jar!/:5.3.9]
		at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:688) [spring-aop-5.3.9.jar!/:5.3.9]

ava.lang.NullPointerException: null
	at com.azure.resourcemanager.compute.implementation.VirtualMachineScaleSetImpl.initializeChildrenFromInner(VirtualMachineScaleSetImpl.java:183) ~[azure-resourcemanager-compute-2.8.0.jar!/:2.8.0]
	at com.azure.resourcemanager.resources.fluentcore.arm.models.implementation.GroupableParentResourceImpl.<init>(GroupableParentResourceImpl.java:32) ~[azure-resourcemanager-resources-2.8.0.jar!/:2.8.0]
	at com.azure.resourcemanager.compute.implementation.VirtualMachineScaleSetImpl.<init>(VirtualMachineScaleSetImpl.java:169) ~[azure-resourcemanager-compute-2.8.0.jar!/:2.8.0]
	at com.azure.resourcemanager.compute.implementation.VirtualMachineScaleSetsImpl.wrapModel(VirtualMachineScaleSetsImpl.java:238) ~[azure-resourcemanager-compute-2.8.0.jar!/:2.8.0]
	at com.azure.resourcemanager.compute.implementation.VirtualMachineScaleSetsImpl.wrapModel(VirtualMachineScaleSetsImpl.java:32) ~[azure-resourcemanager-compute-2.8.0.jar!/:2.8.0]
	at com.azure.resourcemanager.resources.fluentcore.arm.collection.implementation.ReadableWrappersImpl.lambda$wrapPageAsync$1(ReadableWrappersImpl.java:32) ~[azure-resourcemanager-resources-2.8.0.jar!/:2.8.0]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[?:1.8.0_292]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384) ~[?:1.8.0_292]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[?:1.8.0_292]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:1.8.0_292]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[?:1.8.0_292]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_292]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566) ~[?:1.8.0_292]
	at com.azure.resourcemanager.resources.fluentcore.utils.PagedConverter.lambda$mapPagedResponse$6(PagedConverter.java:111) ~[azure-resourcemanager-resources-2.8.0.jar!/:2.8.0]
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:106) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxTake$TakeSubscriber.onNext(FluxTake.java:128) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxExpand$ExpandBreathSubscriber.onNext(FluxExpand.java:118) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onNext(FluxSwitchIfEmpty.java:74) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxMap$MapConditionalSubscriber.onNext(FluxMap.java:220) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onNext(MonoFlatMap.java:249) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onNext(FluxSwitchIfEmpty.java:74) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onNext(MonoFlatMap.java:249) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onNext(MonoFlatMap.java:249) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2398) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onSubscribe(MonoFlatMap.java:238) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:55) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:157) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onNext(FluxSwitchIfEmpty.java:74) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2398) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set(Operators.java:2194) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onSubscribe(Operators.java:2068) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:55) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:64) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:157) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoCacheTime$CoordinatorSubscriber.signalCached(MonoCacheTime.java:337) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoCacheTime$CoordinatorSubscriber.onNext(MonoCacheTime.java:354) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:151) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2398) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onSubscribe(MonoFlatMap.java:110) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:55) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:64) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoCacheTime.subscribeOrReturn(MonoCacheTime.java:143) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:57) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:157) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.onNext(FluxDoOnEach.java:173) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:151) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:151) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:151) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.complete(MonoIgnoreThen.java:284) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onNext(MonoIgnoreThen.java:187) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:151) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at reactor.core.publisher.MonoCreate$DefaultMonoSink.success(MonoCreate.java:160) ~[reactor-core-3.4.9.jar!/:3.4.9]
	at com.azure.core.http.okhttp.OkHttpAsyncHttpClient$OkHttpCallback.onResponse(OkHttpAsyncHttpClient.java:181) ~[azure-core-http-okhttp-1.7.3.jar!/:1.7.3]
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519) ~[okhttp-4.9.1.jar!/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
	Suppressed: java.lang.Exception: #block terminated with an error
		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:99) ~[reactor-core-3.4.9.jar!/:3.4.9]
		at reactor.core.publisher.Flux.blockLast(Flux.java:2644) ~[reactor-core-3.4.9.jar!/:3.4.9]
		at com.azure.core.util.paging.ContinuablePagedByIteratorBase.requestPage(ContinuablePagedByIteratorBase.java:94) ~[azure-core-1.20.0.jar!/:1.20.0]
		at com.azure.core.util.paging.ContinuablePagedByItemIterable$ContinuablePagedByItemIterator.<init>(ContinuablePagedByItemIterable.java:50) ~[azure-core-1.20.0.jar!/:1.20.0]
		at com.azure.core.util.paging.ContinuablePagedByItemIterable.iterator(ContinuablePagedByItemIterable.java:37) ~[azure-core-1.20.0.jar!/:1.20.0]
		at com.azure.core.util.paging.ContinuablePagedIterable.iterator(ContinuablePagedIterable.java:106) ~[azure-core-1.20.0.jar!/:1.20.0]

**To Reproduce**
Calling SDK methods:
resourceManager.virtualMachineScaleSets().list();
and 
scaleSet.virtualMachines().list();

***Code Snippet***
AzureResourceManager resourceManager = AzureResourceManager.configure().withHttpClient(httpClient)
				.withScope(AzureEnvironment.AZURE).authenticate(tokenCredential, profile)
				.withDefaultSubscription();

for (VirtualMachineScaleSet scaleSet : resourceManager.virtualMachineScaleSets().list()) {
}

for (VirtualMachineScaleSet scaleSet : resourceManager.virtualMachineScaleSets().list()) {
 for (VirtualMachineScaleSetVM vm : scaleSet.virtualMachines().list()) {
  }
}

Add the code snippet that causes the issue.

**Expected behavior**
A VMScaleSet and VM list should be returned.
**Screenshots**

**Setup (please complete the following information):**
 - OS: Linux
 - IDE: NA
 - Library/Libraries: azure-resourcemanager-compute-2.8.0.jar!/:2.8.0
 - Java version: 8
 - App Server/Environment: Tomcat
 - Frameworks: Spring Boot

If you suspect a dependency version mismatch (e.g. you see `NoClassDefFoundError`, `NoSuchMethodError` or similar), please provide
- verbose dependency tree (`mvn dependency:tree -Dverbose`)
  
**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

fix #25231
Current vmss service support creating vmss with flexible mode and null vmss profile. Adapt sdk accordingly.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetImpl.java [hunk 1]
```diff
@@ -180,7 +180,7 @@
     @Override
     protected void initializeChildrenFromInner() {
         this.extensions = new HashMap<>();
-        if (this.innerModel().virtualMachineProfile().extensionProfile() != null) {
+        if (this.innerModel() != null && this.innerModel().virtualMachineProfile() != null && this.innerModel().virtualMachineProfile().extensionProfile() != null) {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetImpl.java:183]: line maybe too long.

**Review by weidongxu-microsoft — APPROVED** [review]
Please merge latest main, and run the live tests in compute.

**Comment by weidongxu-microsoft** [comment]
Will it help to add the `orchestrationMode` property to VMSS interface?

**Comment by XiaofeiCao** [comment]
> Will it help to add the `orchestrationMode` property to VMSS interface?

Sure, since it's missing in the interface.

**Review by haolingdong-msft — APPROVED** [review]

**Comment by XiaofeiCao** [comment]
Closing this pr due to mis-operation. Opening another pr.

**Pull request closed without merging**